packages/astro/src/assets/fonts/providers/local.ts

```diff
@@ -14,43 +14,58 @@ type InitializedProvider = NonNullable<Awaited<ReturnType<unifont.Provider>>>;
 
 type ResolveFontResult = NonNullable<Awaited<ReturnType<InitializedProvider['resolveFont']>>>;
 
-interface Options {
+interface ResolveOptions {
 	root: URL;
+	proxyURL: (value: string) => string;
 }
 
-interface ResolveOptions {
-	proxyURL: URLProxy;
-}
+export function resolveLocalFont(
+	family: LocalFontFamily,
+	{ proxyURL, root }: ResolveOptions,
+): ResolveFontResult {
+	const fonts: ResolveFontResult['fonts'] = [];
 
-// TODO: dev watcher and ways to update during dev
-export function createLocalProvider({ root }: Options) {
-	return {
-		resolveFont: async (
-			family: LocalFontFamily,
-			{ proxyURL }: ResolveOptions,
-		): Promise<ResolveFontResult> => {
-			const fonts: ResolveFontResult['fonts'] = [];
-
-			for (const src of family.src) {
-				for (const weight of src.weights ?? DEFAULTS.weights) {
-					for (const style of src.styles ?? DEFAULTS.styles) {
-						// TODO: handle fallbacks?
-						// TODO: handle subset
-						fonts.push({
-							weight,
-							style,
-							src: src.paths.map((path) => ({
-								url: proxyURL(fileURLToPath(new URL(path, root))),
-								format: extractFontType(path),
-							})),
-						});
-					}
-				}
+	for (const src of family.src) {
+		for (const weight of src.weights ?? DEFAULTS.weights) {
+			for (const style of src.styles ?? DEFAULTS.styles) {
+				// TODO: handle fallbacks?
+				// TODO: handle subset
+				fonts.push({
+					weight,
+					style,
+					src: src.paths.map((path) => ({
+						url: proxyURL(fileURLToPath(new URL(path, root))),
+						format: extractFontType(path),
+					})),
+				});
 			}
+		}
+	}
 
-			return {
-				fonts,
-			};
-		},
+	return {
+		fonts,
 	};
 }
+
+export class LocalFontsWatcher {
+	getPaths: (() => Array<string>) | null = null;
+	update: (() => void) | null = null;
```

Comment:
If this function can be null, then have the `constructor` accept the `update` callback. Doing `const a = new AClass(); a.update() = something` isn't a good pattern, because it relies on the caller to **mutate** internal functions of the class. 

Same for `getPaths`.

Reply:
For the record, things changed while I coded this thing so this was useful at some point. It's not the case anymore so he now uses a constructor